File: src/main/java/com/github/bpark/companion/DbPediaClient.java
package com.github.bpark.companion;

import org.eclipse.rdf4j.model.Value;
import org.eclipse.rdf4j.query.BindingSet;
import org.eclipse.rdf4j.query.QueryLanguage;
import org.eclipse.rdf4j.query.TupleQuery;
import org.eclipse.rdf4j.query.TupleQueryResult;
import org.eclipse.rdf4j.repository.Repository;
import org.eclipse.rdf4j.repository.RepositoryConnection;
import org.eclipse.rdf4j.repository.sparql.SPARQLRepository;

/**
 * @author ksr
 */
public class DbPediaClient {

    public static final String[] prefixes = {
            "PREFIX owl: <http://www.w3.org/2002/07/owl#>",
            "PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>",
            "PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>",
            "PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>",
            "PREFIX foaf: <http://xmlns.com/foaf/0.1/>",
            "PREFIX dc: <http://purl.org/dc/elements/1.1/>",
            "PREFIX : <http://dbpedia.org/resource/>",
            "PREFIX dbpedia2: <http://dbpedia.org/property/>",
            "PREFIX dbpedia: <http://dbpedia.org/>",
            "PREFIX skos: <http://www.w3.org/2004/02/skos/core#>",
            "PREFIX ont: <http://dbpedia.org/ontology/>"
    };

    public static void main(String[] args) {
        String sparqlEndpoint = "http://dbpedia.org/sparql";
        Repository repo = new SPARQLRepository(sparqlEndpoint);
        repo.initialize();

        String prefix = String.join("\n", prefixes);

        String query = prefix + "\nSELECT ?comment WHERE {\n" +
                "?body a ont:CelestialBody .\n" +
                "?body foaf:name \"Vega\"@en .\n" +
                "?body rdfs:comment ?comment .\n" +
                "FILTER ( lang(?comment) = \"en\")\n" +
                "}";

        System.out.println(query);

        try (RepositoryConnection conn = repo.getConnection()) {
            TupleQuery tupleQuery = conn.prepareTupleQuery(QueryLanguage.SPARQL, query);
            try (TupleQueryResult result = tupleQuery.evaluate()) {
                while (result.hasNext()) {  // iterate over the result
                    BindingSet bindingSet = result.next();
                    Value comment = bindingSet.getValue("comment");
                    System.out.println(comment.stringValue());
                }
            }
        }

    }
}

File: src/main/kotlin/com/github/bpark/companion/DbPediaClient.kt
package com.github.bpark.companion

import org.eclipse.rdf4j.query.QueryLanguage
import org.eclipse.rdf4j.repository.sparql.SPARQLRepository

fun main(args: Array<String>) {

    val prefixes = arrayOf("PREFIX owl: <http://www.w3.org/2002/07/owl#>",
            "PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>",
            "PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>",
            "PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>",
            "PREFIX foaf: <http://xmlns.com/foaf/0.1/>",
            "PREFIX dc: <http://purl.org/dc/elements/1.1/>",
            "PREFIX : <http://dbpedia.org/resource/>",
            "PREFIX dbpedia2: <http://dbpedia.org/property/>",
            "PREFIX dbpedia: <http://dbpedia.org/>",
            "PREFIX skos: <http://www.w3.org/2004/02/skos/core#>",
            "PREFIX ont: <http://dbpedia.org/ontology/>")

    val sparqlEndpoint = "http://dbpedia.org/sparql"
    val repo = SPARQLRepository(sparqlEndpoint)
    repo.initialize()

    val prefix = prefixes.joinToString("\n")

    val query = prefix + "\nSELECT ?comment WHERE {\n" +
            "?body a ont:CelestialBody .\n" +
            "?body foaf:name \"Vega\"@en .\n" +
            "?body rdfs:comment ?comment .\n" +
            "FILTER ( lang(?comment) = \"en\")\n" +
            "}"

    println(query)

    repo.connection.use { conn ->
        val tupleQuery = conn.prepareTupleQuery(QueryLanguage.SPARQL, query)
        tupleQuery.evaluate().use { result ->
            while (result.hasNext()) {  // iterate over the result
                val bindingSet = result.next()
                val comment = bindingSet.getValue("comment")
                println(comment.stringValue())
            }
        }
    }
}